Add support for jQuery 4.0.0-beta.2

jayaddison opened the pull request:
This pull request includes a

- [ ] Bug fix
- [x] New feature
- [ ] Translation

The following changes were made

- Prepares support for jQuery 4.0.0 by testing against the second beta of that release (as [announced on the jQuery blog](https://blog.jquery.com/2024/07/17/second-beta-of-jquery-4-0-0/)).
- Builds on work by @marcellov7 to add unit and integration test coverage using [`jquery-4.0.0-beta.2`](https://github.com/jquery/jquery/commit/51fffe9f7395f86fb24c59115c9b98855c39fc07).
- Fixes a small bug in the `$.camelCase` replacement implementation.

If this is related to an existing ticket, include a link to it as well.

Relates to (but does not resolve) #6298.

Edit: remove dependency changes and no-longer-included commits from this description.

marcellov7 committed:
Add support for `jQuery` v4.0.0-beta.2

Co-authored-by: James Addison <james@reciperadar.com>

JCCChris commented on docs/themes/site/js/data-fill-from.js:
```diff
@@ -5,7 +5,7 @@ $(document).ready(function() {
     var $container = $(codeContainer);
 
-    var code = $.trim($container.html());
+    var code = $container.html().trim();
```
https://api.jquery.com/jQuery.trim/

$.trim() handles nulls and undefined.  Should this also handle that?

jayaddison replied:
Yep, that seems worthwhile - it does seem that `.html()` can return `undefined` if the container element is not found in the DOM.

jayaddison committed:
Documentation: retain defensive coding in retrieval of code content from HTML elements.

jayaddison commented on .github/workflows/main.yml:
```diff
@@ -20,14 +20,19 @@ jobs:
       - uses: actions/checkout@v1
-      - name: Use Node.js 12
+      - name: Use Node.js 18
         uses: actions/setup-node@v1
         with:
-          node-version: 12
+          node-version: 18
```
Two notes about the suggested Node v18 upgrade here:

  * Currently, Node v20 (`iron`) is the long-term-support (LTS) version of Node - so that might be a better choice here.
  * These dependency upgrades should potentially be isolated into a separate pull request (rationale: change one aspect of a project at a time, to identify as much possible breakage -- because then we can resolve that -- as possible).

I'll place this pull request into draft status until the dependencies are updated.

jayaddison committed:
Build: recompile distributable files

jayaddison commented:
I think these changes are ready for re-review; however I'm also planning to perform a git rebase to rearrange them into a more coherent series of commits.

jayaddison committed:
Tests: update `jquery-mockjax` to v2.7.0-beta.0

This beta/prerelease version of `jquery-mockjax` includes jQuery-4.0.0.beta2 support.

See https://github.com/jakerella/jquery-mockjax/issues/362 for details.

jayaddison commented:
One mistaken CSS merge conflict resolution in 4494a73cfe5f5d07cc5cb1ebe84afd931ed44886 - fixing that momentarily..

jayaddison commented:
cc @kevin-brown - this PR collates some work from #6298 to add support for the beta versions of jQuery 4.0.0 - note that an update to a corresponding beta version of `jquery-mockjax` is also required because it required some compatibility updates too.

kevin-brown commented:
Thanks a bunch for getting this one updated, it's definitely a lot easier to review now that the other QUnit-related pull requests have also been merged. 😄
